src/Installer/Installer/Factory.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Dock\Installer\Installer;
+
+use Dock\System\OS;
+
+class Factory
+{
+    public function getInstaller(OS $os)
+    {
+        $operatingSystem = $os->get();
+
+        switch ($operatingSystem) {
```

Comment:
I thought this was a textbook use of a switch. Plus it allows us to include the name of the unsupported OS in the exception, which can be useful when debugging. What would be the benefits of your solution? It's the first time I see something like that!